Added 'region' and 'insecure' flag

DustinKLo opened the pull request:
related issues: 
- https://github.com/abutaha/aws-es-proxy/issues/68
- https://github.com/abutaha/aws-es-proxy/issues/83

the `region` and `service` values are parsed from the `endpoint` flag but sometimes the url is inconsistent with how the code is expecting it, so the `region` value is an empty string which returns a `403`
```
level=error msg="Received 403 from AWSAuth, invalidating credentials for retrial"
```
now the proxy has a optional `region` flag (ie. `-region us-west-2`)

the proxy also assumes that your server has the proper SSL certificates and will return a verification error when you don't
i added a optional `insecure` flag to fix that issue
```
level=error msg="Get \"https://elastic.example.comt/\": x509: certificate is valid for *.eu-central-1.es.amazonaws.com, not elastic.example.com
```

DustinKLo committed:
added region flag to avoid having to parse the endpoint to get the region (#1)

hard-coded proxy.service value to "es"
added insecure flag and logic to avoid SSL verification

DustinKLo committed:
Code cleanup (#2)

* added region flag to avoid having to parse the endpoint to get the region

hard-coded proxy.service value to "es"
added insecure flag and logic to avoid SSL verification

* cleaning up code, moving insecure logic to main function (TODO: still need to test)

* removed redundant use of insecure flag

DustinKLo committed:
squash commits

DustinKLo committed:
fixed merge conflicts with upstream/master

abutaha commented:
Thank you @DustinKLo. I like the SSL insecure option, however, I'm a bit concerned about having 'region' as a mandatory option and changing the current behavior of the proxy as this change will break many implementations of the proxy across multiple users who have the proxy installed and configured as part of their CI/CD. This change will break it for them and I want to keep backward compatibility within v1.x.

With that said, I've started the development of v2.x which WILL break the backward compatibility. In v2.x I'm moving command line options and parameters to a config file to make it easier to update the settings. I've included insecure ssl and region in the config in v2.0. Please have a look and let me know if you have any ideas.

Thanks,

DustinKLo commented:
@abutaha thanks for the review 👍 

i can make the `region` flag optional and re-add the `endpoint` parsing logic if the `region` flag isn't supplied in the command line
this should ensure backwards compatibility I hope 🤞

Let me know what you think, i can continue development if you want to go this direction

abutaha commented:
> @abutaha thanks for the review 
> 
> i can make the `region` flag optional and re-add the `endpoint` parsing logic if the `region` flag isn't supplied in the command line
> this should ensure backwards compatibility I hope 
> 
> Let me know what you think, i can continue development if you want to go this direction

Yes please go ahead. Just ensure to pull the latest master.

Thanks,

DustinKLo committed:
Fix merge conflict (#3)

* Added support for assuming a role that has access to ElasticSearch

* Fixed small conflicts

* Added new contributors

Co-authored-by: Joshua Garnett <josh.garnett@gmail.com>
Co-authored-by: Dido (Christoph Poelt) <christoph.poelt@valiton.com>
Co-authored-by: Muslim AbuTaha <muslim.adel@gmail.com>

DustinKLo committed:
re-added endpoint parsing login (originally removed and replced with a region flag) (#4)

made -region flag optional, if not supplied will go through the endpoint parsing logic instead
edited README.md description for region flag
added "assume" description to --help in README.md

DustinKLo committed:
fixed merge conflicts

DustinKLo commented:
@abutaha thanks for the response
I re-added the endpoint parsing logic and also allows for a optional `region` flag
please review when you have the time 👍

DustinKLo committed:
small code formatting

DustinKLo committed:
fixed parseEndpoint() bug where p.region should be empty string to parse endpoint

pshomov commented:
This seems like something that is ready to be merged but comms ended abruptly. Sure would appreciate having this fixed 👍🏻.Thank you for the great work on this tool 🙏🏻 .

DustinKLo committed:
Merge branch 'abutaha:master' into master

neuralsandwich commented:
I've been experiencing the 403 as well but testing the changes on `hysds:master` doesn't seem to have addressed the issue.

Commit:
merge upstream/master
